packages/odyssey-react/src/components/TextInput/TextInput.tsx

```diff
@@ -143,47 +164,54 @@ export const TextInput = withTheme(
           }
         : {};
 
-    const input = (
-      <input
-        {...omitProps}
-        {...ariaProps}
-        className={styles.root}
-        disabled={disabled}
-        id={oid}
-        name={name}
-        onChange={handleChange}
-        onBlur={onBlur}
-        onFocus={onFocus}
-        placeholder={placeholder}
-        readOnly={readonly}
-        ref={ref}
-        required={required}
-        type={type}
-        defaultValue={defaultValue}
-        value={value}
-      />
-    );
-
-    const search = (
-      <span className={styles.outer}>
-        {input}
-        <span className={styles.indicator} role="presentation">
-          <SearchIcon />
-        </span>
-      </span>
-    );
-
     return (
       <Field
         error={error}
         hint={hint}
         inputId={oid}
         label={label}
-        labelHidden={type === "search"}
+        labelHidden={isSearchTextInput}
         optionalLabel={optionalLabel}
         required={required}
       >
-        {type === "search" ? search : input}
+        <div className={styles.root}>
+          {(prefix || isSearchTextInput) && (
```

Comment:
If `type="password"` then we can hijack the `suffix` like we do here with the prefix/search.
But if we need to control the suffix/prefix when type is text or generically then this solution won't work. So I don't know if what is in this PR is acceptable or not but at least we can discuss from here?

Reply:
hmmm I can understand the need for future work for `type="password"`, but I'm not sure I follow the requirement around controlling prefixes/suffixes for `type="text"` ? Is there some nuance here I've missed for the intended outcome?

Reply:
I've heard talk about a `copyInput`? That would need an icon button as the suffix. The way `TextInput` is written now, I don't know we would achieve that?